src/main/scala/io/iohk/ethereum/jsonrpc/TransactionRequest.scala

```diff
@@ -20,8 +20,19 @@ case class TransactionRequest(
       nonce = nonce.getOrElse(defaultNonce),
       gasPrice = gasPrice.getOrElse(defaultGasPrice),
       gasLimit = gasLimit.getOrElse(defaultGasLimit),
-      receivingAddress = to,
+      receivingAddress = if (to.contains(Address(0))) None else to,
```

Comment:
It seems this is the default behavior in cpp-ethereum (and what solidity tests expect). To me it (kind of) makes sense, but is a bit counterintuitive for caller (what if you really want to send tx to address 0x0?). WDYT? Other solution is to use this logic only in testmode.

Reply:
Well, you know... https://gastracker.io/addr/0x0000000000000000000000000000000000000000

This is of course a result of many mistakes, but maybe, for reasons beyond my imagination, someone has interest in sending Ether there. I'd say if it's not too much trouble go with testmode only, if not, don't worry about it.

Btw, the expression could be simplified as:
```
to.filter(_ != Address(0))
```